set up landing page

alexeykuz-sys opened the pull request:
i have written description somewhere else:). Sorry to be complete pain...
Landing page: created accordion and carousel divs to represent  future and past hackathons. I checked #hackathon tab in slack but havent found any suitable text to fill up accordion and would like to discuss. Meantime, basic setup, decoration and fonts are in accordance with wireframes. i checked with validators and made sure the page is responsive. Please let me know if i need to do anything else. Thank you
![Screen Shot 2020-10-21 at 12 50 45](https://user-images.githubusercontent.com/64777928/96724060-5024fe80-13a7-11eb-8640-c416cfbd0811.png)
![Screen Shot 2020-10-21 at 12 53 10](https://user-images.githubusercontent.com/64777928/96724062-51eec200-13a7-11eb-9581-722578b860ae.png)

stefdworschak commented:
@alexeykuz-sys could you please add some context to the Pull Request (e.g. Description of what it should do, which User Story this is connected to, what testing you've done, Screenshots if there is a design element, etc.)?

stefdworschak commented on index.html:
```diff
@@ -0,0 +1,115 @@
+<!DOCTYPE html>
```
Could you please use the base template at `templates/base.html` and extend that with your body content instead of creating the whole structure?

alexeykuz-sys replied:
done.

stefdworschak commented on index.html:
```diff
@@ -0,0 +1,115 @@
+
+                    <div class="collapse-content">
+
+                    <!-- Text -->
```
Could you please remove this and all unnecessary comments below (e.g. `<!-- Text -->`, `<!-- Button -->`)?

alexeykuz-sys replied:
removed all comments and spaces.

stefdworschak commented on index.html:
```diff
@@ -0,0 +1,115 @@
+
+                 </div>
+
+                <!---<div class="card-body">
```
Please remove unneeded code that is commented out.

alexeykuz-sys replied:
done

stefdworschak reviewed (changes requested):
@alexeykuz-sys in general I would review what kind of content goes on the home page and maybe first draw up a Wireframe. This kind of list of Hackathons has already been covered in [#P04. As Participant, I would like to see a list of any upcoming Hackathons.](https://github.com/Code-Institute-Community/ci-hackathon-app/issues/21). 

Also, from the looks of it your fork seems to be a bit behind master, so I would use `git pull upstream master` to update it as well.

TravelTimN commented on templates/base.html:
```diff
@@ -37,15 +38,15 @@
-<script src="https://code.jquery.com/jquery-3.5.1.slim.min.js"
-    integrity="sha384-DfXdz2htPH0lsSSs5nCTpuj/zy4C+OGpamoFVy38MVBnE+IbbVYUew+OrCXaRkfj" crossorigin="anonymous">
-</script>
+<script src="https://code.jquery.com/jquery-1.12.4.js"></script>
```
What is the purpose of going back from jQuery version 3.5.1 over to a much older version 1.12.4?

alexeykuz-sys replied:
we moved from jquery to bootstrap and resolve this issue.

stefdworschak replied:
@alexeykuz-sys can we change it to this please `<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>`. That's a newer version, but not slim so it should do the trick.

TravelTimN commented on templates/base.html:
```diff
@@ -37,15 +38,15 @@
 <script src="https://cdn.jsdelivr.net/npm/bootstrap@4.5.3/dist/js/bootstrap.min.js"
     integrity="sha384-w1Q4orYjBQndcko6MimVbzY0tgp4pWB4lZ7lr30WKz0vr/aWKhXdBNmNb5D92v7s" crossorigin="anonymous">
 </script>
+
```
This blank line doesn't need to be here, it's so any custom <script> elements can be injected directly after the base files.

alexeykuz-sys replied:
noted

TravelTimN commented on templates/base.html:
```diff
@@ -37,15 +38,15 @@
 <script src="https://cdn.jsdelivr.net/npm/popper.js@1.16.1/dist/umd/popper.min.js"
     integrity="sha384-9/reFTGAW83EW2RDu2S0VKaIzap3H66lZH81PoYlFhbGU+6BZp6G7niu735Sk7lN" crossorigin="anonymous">
 </script>
+ <script src="https://code.jquery.com/ui/1.12.1/jquery-ui.js"></script>
```
Can you provide some examples of the use of jQueryUI used on this project? Is this needed?

alexeykuz-sys replied:
we changed from jquery to bootstrap.

TravelTimN commented on templates/base.html:
```diff
@@ -12,6 +12,7 @@
         integrity="sha512-1PKOgIY59xJ8Co8+NE6FZ+LOAZKjy+KY8iq0G4B3CyeY6wYHN3yt9PW0XpSriVlkMXe40PTKnXrLnZ9+fkDaog=="
         crossorigin="anonymous">
     <link href="https://fonts.googleapis.com/css2?family=Montserrat&family=Rubik&display=swap" rel="stylesheet">
+    <link rel="stylesheet" href="https://code.jquery.com/ui/1.12.1/themes/base/jquery-ui.css">
```
Can you provide some examples of the use of jQueryUI used on this project? Is this needed, or are we fine to just use the normal jQuery?

TravelTimN commented on static/js/script.js:
```diff
@@ -0,0 +1,3 @@
+$(document).ready(function(){
+    $( "#accordion" ).accordion();
+});
```
Can you just check this jQuery code against an online Beautify tool?
Some spaces where they shouldn't be, and missing spaces where there should be.
Good example: https://beautifier.io/

TravelTimN commented on static/css/style.css:
```diff
@@ -156,4 +156,7 @@ footer ul li a {
\ No newline at end of file
+}
+
+/* Landing Page */
```
Was there any custom CSS styles added to the landing page, or was this comment optional just in case? If no custom styles, I don't think we need a blank comment here then.

TravelTimN commented on home/templates/home/index.html:
```diff
@@ -3,10 +3,88 @@
+
+        
+    </div><!--row div-->
+</div><!--container div-->
```
We don't need the code-comments for everything. Can you remove these? (and all of the ones above from this file)

TravelTimN commented on home/templates/home/index.html:
```diff
@@ -3,10 +3,88 @@
+                    <img src="..." class="d-block w-50" alt="...">
+                    </div>
+                </div>
+                <a class="carousel-control-prev" href="#carouselExampleControls" role="button" data-slide="prev">
```
Let's try to make our href and ID unique to the actual carousel itself. It could be something like `#carousel-project-showcase` for example.

TravelTimN commented on home/templates/home/index.html:
```diff
@@ -3,10 +3,88 @@
+        </div><!---col div-->
+    <!--carousel-->  
+        <div class = "col-6 col-sm-12  col-md-6">
+            <div id="carouselExampleControls" class="carousel slide" data-ride="carousel">
```
Again, let's try to give the IDs something more unique to our project, perhaps something like: `id="carousel-project-showcase"`.

TravelTimN commented on home/templates/home/index.html:
```diff
@@ -3,10 +3,88 @@
+            <div id="carouselExampleControls" class="carousel slide" data-ride="carousel">
+                <div class="carousel-inner">
+                    <div class="carousel-item active">
+                    <img src="" class="d-block w-50" alt="...">
```
Let's get some visual context here, you can use some of the images from the static folder for sample carousel display, so they're not broken images/links.

TravelTimN commented on home/templates/home/index.html:
```diff
@@ -3,10 +3,88 @@
+            </div>
+        </div><!---col div-->
+    <!--carousel-->  
+        <div class = "col-6 col-sm-12  col-md-6">
```
Check the spaces here, and also, can you confirm this is correct for the sizes?
`col-6 col-sm-12 col-md-6` it seems a bit redundant having 6 twice, no?

alexeykuz-sys replied:
do you mean col-6 and col-md-6?

TravelTimN commented on home/templates/home/index.html:
```diff
@@ -3,10 +3,88 @@
+                    </p>
+                </div>
+            </div>
+        </div><!---col div-->
```
Remember to remove these random code-comments.

alexeykuz-sys replied:
done

TravelTimN commented on home/templates/home/index.html:
```diff
@@ -3,10 +3,88 @@
+            <h3>Hackathon 01 Title</h3>
+                <div>
+                    <p>
+                    Mauris mauris ante, blandit et, ultrices a, suscipit eget, quam. Integer
```
Even if it's fake Hackathon data, something a bit more suitable to the project would be great, instead of Lorem text for example, Would you mind providing some "fake" Hackathon details. Feel free to checkout the #hackathons channel on Slack and just use the details from a previous Hackathon to get some inspiration of details (dates/times/theme/etc.).

alexeykuz-sys replied:
ok will do it now and update!

TravelTimN commented on home/templates/home/index.html:
```diff
@@ -3,10 +3,88 @@
-    <small>custom class: <strong>btn-ci</strong></small>
+   <div class="container-fluid">
+    <div class="row">
+        <div class = "col-6 col-sm-12  col-md-6">
```
check the extra spaces here, they're not needed.
There are trailing spaces on several places, you might consider putting the HTML code into an online code-formatter, just to get it formatted/cleaned up.
Also, is this correct or a bit redundant?
`col-6 col-sm-12 col-md-6` --- do we need 6 twice here?

TravelTimN reviewed (changes requested):
Hi @alexeykuz-sys - please see some items on this Review that could be addressed. Thanks! Let us know if you have any questions.

alexeykuz-sys commented:
made changes. deleted jquery ui and updated jquery with latest version. i deleted as it was left in the code when i planned to use jquery for accordion. moved to bootstrap later. also sorted style.css conflict. Hope all correct now? thanks

stefdworschak reviewed (changes requested):
@alexeykuz-sys could you just have a look at the merge-conflict that has not been resolved yet? Once that's done I can approve the PR.

alexeykuz-sys commented:
Resolved merge conflict.

alexeykuz-sys commented on static/css/style.css:
```diff
@@ -1,3 +1,242 @@
+<<<<<<< HEAD
```
resolved merge conflict! thanks

stefdworschak replied:
@alexeykuz-sys I don't think that this is resloved? I can stilll see the `<<<<<<< HEAD` here and there seems to be a merge conflict in the `base.html` file now too.

stefdworschak reviewed (approved):
LGTM
